tests/test_plot.py

```diff
@@ -300,12 +300,20 @@ def test_2d_colorbar_options(function, colorbar, handsome_signal_2d):
     fig = create_figure()
     if colorbar == "off":
         # test not plotting a colorbar
-        function(signal, colorbar=False)
+        if function == plot.spectrogram:
+            with pytest.warns(UserWarning, match="Using only the first"):
+                function(signal, colorbar=False)
+        else:
+            function(signal, colorbar=False)
     elif colorbar == "axes":
         # test plotting colorbar to specified axis
         fig.clear()
         _, ax = plt.subplots(1, 2, num=fig.number)
-        function(signal, ax=ax)
+        if function == plot.spectrogram:
+            with pytest.warns(UserWarning, match="Using only the first"):
+                function(signal, colorbar=False)
```

Comment:
using `function(signal[0], colorbar=False)` would solve the issue more explicitly

Reply:
yes, changed that